### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by Steve888888** [koalaman/shellcheck#1964]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2148, SC1083, SC2154, SC2086, SC2154, SC2086, SC1083, SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
echo {"gp_ct_planned_cases":$testcase_num, "gp_ct_passed_cases":$testcase_succ} > $WORKSPACE/report.json
```

#### Here's what shellcheck currently says:
```
Line 1	SC2148: Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
Line 1	SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
Line 1	SC2154: testcase_num is referenced but not assigned.
Line 1	SC2086: Double quote to prevent globbing and word splitting.
Line 1	SC2154: testcase_succ is referenced but not assigned.
Line 1	SC2086: Double quote to prevent globbing and word splitting.
Line 1	SC1083: This } is literal. Check expression (missing ;/\n?) or quote it.
Line 1	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by brother** [comment]
If you follow the advise I am pretty sure the code will be nice.

No errors reported (when shell was set to bash, first warning...)
`echo "{gp_ct_planned_cases:$testcase_num, gp_ct_passed_cases:$testcase_succ}" > "$WORKSPACE/report.json"`

**Comment by koalaman** [comment]
You appear to be using JSON notation directly in Bash as you would in JavaScript. It only appears to work in this case because the value is sufficient trivial, but e.g. a string value would break it. You should quote as demonstrated.
